### PR TITLE
Bug Fix Snmp empty metric name

### LINKED ISSUE #8840: SNMP input can produce metrics with empty metric name
### Steps to reproduce:
The snmp input can have a table with no oid or name but with fields.  (see the plugin readme https://github.com/influxdata/telegraf/blob/ea4feb1a07edfea3243cc328f1994f63852e9f96/plugins/inputs/snmp/README.md#table)  If the table name and the table oid are not set, the plugin produces metrics with no name.

### Relevant telegraf.conf:
Here's an example snmp input that exhibits the problem.
```toml
[[inputs.snmp]]
  agents = ["udp://127.0.0.1:161"]
  community = "public"
  version = 2
  [[inputs.snmp.table]]
    #oid = "IF-MIB::ifXTable"
    #name = "ifXTable"
    index_as_tag = true
    [[inputs.snmp.table.field]]
      oid = "IF-MIB::ifHCInOctets"
      is_tag = false
    [[inputs.snmp.table.field]]
      oid = "IF-MIB::ifHCOutOctets"
      is_tag = false
[[outputs.file]]
  files = ["stdout"]
  data_format = "influx"

```
For this example you need a snmp agent running on localhost that allows access to IF-MIB::ifXTable with the public community string.  I used net-snmp.

If you uncomment either the name or the oid settings, the metric is named so metrics are output.

### System info:
I tested with telegraf master on linux but this is not platform specific.

### Expected behavior:
I didn't expect telegraf to try to produce a metric with no name.  Once I understood this was happening I would have expected an error from the snmp input or the agent itself or the file output.

The snmp input should provide a specific error, like "error: SNMP table at telegraf.conf line 234 is not named.  One or both of the oid and name settings must be set".

### Actual behavior:

Telegraf doesn't produce any metrics

### PR DESCRIPTION
**Pull request by MyaLongmire**
### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.

resolves #8840  and pull request #9366 

Gave name to ifName processor

### CODE REVIEW

**Comment by MyaLongmire** [comment]
@Hipska Will you please test this with ifName just to double check?

**Review by srebhan — COMMENTED** [review]
Hey @MyaLongmire thanks for looking into this!
However, I have some concerns relaxing the conditions here as it opens the door for configuration errors... I would prefer to set `t.Name` to something non-empty and leave `t.Oid` empty _OR_ have an explicit flag to ignore that condition that can be used by the `ifName` processor.

**Comment by MyaLongmire** [comment]
> Hey @MyaLongmire thanks for looking into this!
> However, I have some concerns relaxing the conditions here as it opens the door for configuration errors... I would prefer to set `t.Name` to something non-empty and leave `t.Oid` empty _OR_ have an explicit flag to ignore that condition that can be used by the `ifName` processor.

@srebhan I have changed it back to an error. I also gave the ifName processor a name so it will not error. Thank you for looking this over!

**Review by srebhan — APPROVED** [review]
Wow that's really smart! Looks nice to me. Thanks for squashing this one. :-)

**Comment by srebhan** [comment]
Oh @MyaLongmire, please adapt the PR description as it no longer matches your fix.

File: plugins/processors/ifname/ifname.go [hunk 1]
```diff
@@ -349,6 +349,7 @@ func init() {
 func makeTableNoMock(fieldName string) (*si.Table, error) {
 	var err error
 	tab := si.Table{
+		Name:       "ifName",
```
Comment (Hipska) [inline plugins/processors/ifname/ifname.go:352]:
```suggestion
		Name:       "ifTable",
```

Looks good to me, just would pick a more correct name..

**Review by Hipska — APPROVED** [review]

File: plugins/processors/ifname/ifname.go (hunk 1 above)
Comment (MyaLongmire) [reply plugins/processors/ifname/ifname.go:352]: How is ifTable more correct? Especially since it is the ifName processor.

**Review by helenosheaa — APPROVED** [review]
LGTM

File: plugins/processors/ifname/ifname.go (hunk 1 above)
Comment (reimda) [reply plugins/processors/ifname/ifname.go:352]: ifTable is the snmp table where you read the interface name

**Pull request merged**